### ISSUE TITLE
0.3.0 Upgrade

### ISSUE THREAD

**Issue by dstana** [paginagmbh/redmine_lightbox2#26]
Redmine is showing I've upgraded the plugin, but I'm not seeing the magnifying glass to use the lightbox. I've tried clicking the thumbnails as well and it doesn't seem to work.

Is there a setting I'm missing?

**Comment by tofi86** [comment]
Please go to Admin -> Info and copy all the environment and plugin information and post it here.

**Comment by dstana** [comment]
Environment:
  Redmine version                3.3.0.stable
  Ruby version                   1.9.3-p551 (2014-11-13) [x86_64-linux]
  Rails version                  4.2.6
  Environment                    production
  Database adapter               Mysql2
SCM:
  Git                            1.7.1
  Filesystem                     
Redmine plugins:
  redmine_lightbox2              0.3.0

**Comment by tofi86** [comment]
Okay, this should work. I have the same environment for testing...

If you don't see the magnifier icon which should be provided by the redmine theme, I guess you're using a non-default theme, right? Does the theme have support for redmine 3.3?
There have been big changes to the theme engine for icon support in Redmine 3.3: http://www.redmine.org/issues/21256

**Comment by dstana** [comment]
Ok, the theme is the issue for the magnifier icon but it still isn't working when I switch to the default theme. The image opens up within Redmine (menus and such are still visible instead of directing to an image only page) but the lightbox doesn't work.

**Comment by tofi86** [comment]
Did you follow the installation instruction for the plugin? Did you run the following command after upgrading the plugin source
```
rake redmine:plugins:migrate RAILS_ENV=production
```
This would update the javascript sources in `/public/plugin_assets/` and if this step is missing, the plugin won't work.

Which browser are you using?

What is the link for the magnifier icon?
* this: `http://{host}/attachments/{id}/{image.ext}`
* or this: `http://{host}/attachments/download/{id}/{image.ext}`

**Comment by dstana** [comment]
Yes, I did run the rake command after upgrading the plugin source.

The magnifier link points to http://{host}/attachments/{id}/{image.ext}

**Comment by dstana** [comment]
I just tried reinstalling the plugin and I'm getting this error:

rake aborted!
Mysql2::Error: Access denied for user 'root'@'localhost' (using password: NO)

Maybe this is the issue and plugin_assets aren't being updated?

**Comment by tofi86** [comment]
Ok, but this seems to be an configuration error on your side. Are you sure you're using the PRODUCTION environment? Or are you testing with DEVELOPMENT? Check your database setup in config folder...

**Comment by dstana** [comment]
I'm using the production environment, it's strange because nothing has changed. I just checked the database.yml file and everything is as it should be.

**Comment by tofi86** [comment]
I think I cannot help with the database issue, but the (wrong) magnifier link leads to the conclusion, that the plugin is not being loaded at all. Not only the plugin_assets, but also the plugin view overrides.

**Comment by dstana** [comment]
For some reason, while uninstalling and reinstalling the plugin rake is referencing the wrong location (looks like the 1.8 ruby install remnants). Still working on how to fix it.

**Comment by tofi86** [comment]
Can you upgrade ruby? I just saw that I'm using ruby 2.0.0 in my dev environment.

**Comment by dstana** [comment]
Possibly, but rake is referencing an older version that seems to be hanging around. Still trying to figure out where it calls it from. I'm worried about upgrading it and still having the same issue.

What's really strange is everything else with Redmine seems to operate normally.

**Comment by tofi86** [comment]
> What's really strange is everything else with Redmine seems to operate normally.

Well, redmine or your redmine webserver seems to pick up ruby 1.9.3 correctly (see system information). And only rake seems to pick up wrong ruby version.

Never experienced that, so unfortunately I cannot help :-/

**Comment by dstana** [comment]
So I have to run mine like this for some reason:

`bundle exec rake redmine:plugins:migrate RAILS_ENV=production`

And that does the trick, but the plugin still isn't working. What can I do to check if the plugin_assets have been updated?

**Comment by dstana** [comment]
Any update on this so I can see what's going on?

**Comment by tofi86** [comment]
Sorry for not getting back to you sooner.

Can you check again what the link for the magnifier icon is?
* this: `http://{host}/attachments/{id}/{image.ext}`
* or this: `http://{host}/attachments/download/{id}/{image.ext}`

Can you check your Redmine logfile in `logs/` for errors when loading the issue page?

**Comment by dstana** [comment]
Sorry it's taken awhile to get back, however, I am back with good news. I've managed to resolve this issue. We had a recent crash on one of our servers and everything had to be migrated to a new one (not fun with about 200 pissed off customers that their websites were down).

So Redmine happened to be running on that server and I had to set it up again on the new one. After everything was running smoothly I decided to give lightbox2 a try again. It works! The only thing that's different with the new installation of Redmine is I installed Ruby 2.2 instead of 1.9, perhaps this is the difference?

Thank you so much for your help, great plugin!
